api/client/deals.go

```diff
@@ -20,16 +21,58 @@ type Deals struct {
 
 // WatchEvent is used to send data or error values for Watch.
 type WatchEvent struct {
-	Deal deals.DealInfo
+	Deal deals.StorageDealInfo
 	Err  error
 }
 
+// ListDealRecordsOption updates a ListDealRecordsConfig.
+type ListDealRecordsOption func(*rpc.ListDealRecordsConfig)
+
+// WithFromAddrs limits the results deals initated from the provided wallet addresses.
+func WithFromAddrs(addrs ...string) ListDealRecordsOption {
+	return func(c *rpc.ListDealRecordsConfig) {
+		c.FromAddrs = addrs
+	}
+}
+
+// WithDataCids limits the results to deals for the provided data cids.
+func WithDataCids(cids ...string) ListDealRecordsOption {
+	return func(c *rpc.ListDealRecordsConfig) {
+		c.DataCids = cids
+	}
+}
+
+// WithIncludePending specifies whether or not to include pending deals in the results.
+// Ignored for ListRetrievalDealRecords.
+func WithIncludePending(includePending bool) ListDealRecordsOption {
+	return func(c *rpc.ListDealRecordsConfig) {
+		c.IncludePending = includePending
+	}
+}
+
+// WithOnlyPending specifies whether or not to only include pending deals in the results.
+func WithOnlyPending(onlyPending bool) ListDealRecordsOption {
+	return func(c *rpc.ListDealRecordsConfig) {
+		c.OnlyPending = onlyPending
+	}
+}
+
+// WithAscending specifies to sort the results in ascending order.
+// Default is descending order.
+// If pending, records are sorted by timestamp, otherwise records
+// are sorted by activation epoch then timestamp.
+func WithAscending(ascending bool) ListDealRecordsOption {
+	return func(c *rpc.ListDealRecordsConfig) {
+		c.Ascending = ascending
+	}
+}
+
 // Store creates a proposal deal for data using wallet addr to all miners indicated
 // by dealConfigs for duration epochs.
 func (d *Deals) Store(ctx context.Context, addr string, data io.Reader, dealConfigs []deals.StorageDealConfig, minDuration uint64) ([]cid.Cid, []deals.StorageDealConfig, error) {
 	stream, err := d.client.Store(ctx)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, fmt.Errorf("calling Store: %v", err)
```

Comment:
Trying to start prepending public api errors when I come across them. Nice practice.